kong/plugins/rate-limiting/handler.lua

```diff
@@ -28,10 +28,14 @@ local function get_identifier(conf)
     end
   elseif conf.limit_by == "credential" then
     identifier = ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.id
+  elseif conf.limit_by == "http_header" then
+    identifier = ngx.req.get_headers()[conf.http_header]
```

Comment:
Similar to other proxies, added a fix to use the last header when multiple headers are sent in the request. If you insist on a 400, I can add that but breaking an api seems incorrect.